amazon/ion/ioncmodule.c

```diff
@@ -372,6 +372,34 @@ static iERR ionc_write_sequence(hWRITER writer, PyObject* sequence, PyObject* tu
     Py_DECREF(sequence);
     cRETURN;
 }
+/*
+ * Process and write the key-value pair.
+ *  *  Args:
+ *      writer:  An ion writer
+ *      key: The key of IonStruct item
+ *      val: The value of IonStruct item
+ *      tuple_as_sexp: Decides if a tuple is treated as sexp
+ */
+
+static iERR write_key_value(hWRITER writer, PyObject* key, PyObject* val, PyObject* tuple_as_sexp) {
```

Comment:
nit: name could be clearer. I would call it `write_key_value_pair` or just `write_struct_field` but I read it as "write the value of the key" maybe just me though.

Reply:
> nit: name could be clearer. I would call it `write_key_value_pair` or just `write_struct_field` but I read it as "write the value of the key" maybe just me though.

Thanks for calling this out, `write_struct_field` works better in this case.